src/sentry/incidents/handlers/condition/anomaly_detection_handler.py

```diff
@@ -32,7 +32,7 @@
 
 
 @condition_handler_registry.register(Condition.ANOMALY_DETECTION)
-class AnomalyDetectionHandler(DataConditionHandler[DataPacket]):
+class AnomalyDetectionHandler(DataConditionHandler[dict[str, Any]]):
```

Comment:
could we use a data class or a typeddict instead of `dict[str, Any]`?

It'd be nice to enforce this shape
```
      "values": {
          "value": aggregation_value,
          "source_id": str(self.subscription.id),
          "subscription_id": subscription_update["subscription_id"],
          "timestamp": self.last_update,
      },
```
So we have a known / consistent shape through each method